project/build.scala

```diff
@@ -104,12 +104,35 @@ object build {
       Some(shared(projectBase, conf))
   }
 
+  private val stdOptions = Seq(
+    // contains -language:postfixOps (because 1+ as a parameter to a higher-order function is treated as a postfix op)
+    "-deprecation",
+    "-encoding", "UTF-8",
+    "-feature",
+    "-Xfuture",
+    "-Xsource:2.12",
+    "-Ypartial-unification",
+    "-language:implicitConversions", "-language:higherKinds", "-language:existentials", "-language:postfixOps",
+    "-unchecked"
+  )
+
   private val Scala211_jvm_and_js_options = Seq(
     "-Ybackend:GenBCode",
     "-Ydelambdafy:method",
     "-target:jvm-1.8"
   )
 
+  val lintOptions = Seq(
+    "-Xlint:_,-type-parameter-shadow",
+    "-Yno-adapted-args",
+    "-Ywarn-dead-code",
+    "-Ywarn-unused-import",
+    "-Ywarn-numeric-widen",
+    "-Ywarn-value-discard",
```

Comment:
@puffnfresh what is your opinion on this option? (as you wrote [NonUnitStatements](http://www.wartremover.org/doc/warts.html#nonunitstatements))